molecule/default/tests/test_default.py

```diff
@@ -15,33 +15,35 @@
 @pytest.mark.parametrize("pkg", ["apt-transport-https"])
 def test_apt_https_support(host, pkg):
     """Ensure apt has support for repository URLs using HTTPS."""
-    assert host.package(pkg).is_installed
+    if host.system_info.distribution in ["debian", "kali", "ubuntu"]:
+        assert host.package(pkg).is_installed
 
 
 def test_source_list_for_http(host):
     """Check if the source list file has any URLs still using HTTP."""
-    source_file = "/etc/apt/sources.list"
-    # As of Debian Bullseye the /etc/apt/sources.list file has moved
-    # to /etc/apt/sources.list.d/debian.sources
-    if (
-        host.system_info.distribution == "debian"
-        and host.system_info.codename == "bookworm"
-    ):
-        source_file = "/etc/apt/sources.list.d/debian.sources"
-
-    file_lines = host.file(source_file).content_string.split(os.linesep)
-    sources = [line for line in file_lines if line.startswith("deb")]
-
-    for source in sources:
-        source_elements = source.split(" ")
-        # The second element of a source entry can optionally be a set of options
-        # in the form "[options]".
-        repository_url = source_elements[
-            1 if not source_elements[1].startswith("[") else 2
-        ]
-        if host.system_info.distribution == "ubuntu" and (
-            "ubuntu.com" in repository_url or "canonical.com" in repository_url
+    if host.system_info.distribution in ["debian", "kali", "ubuntu"]:
+        source_file = "/etc/apt/sources.list"
+        # As of Debian Bullseye the /etc/apt/sources.list file has
+        # moved to /etc/apt/sources.list.d/debian.sources
+        if (
+            host.system_info.distribution == "debian"
+            and host.system_info.codename == "bookworm"
         ):
-            # We do not modify the standard repositories on Ubuntu systems
-            continue
-        assert repository_url.startswith("http://") is False
+            source_file = "/etc/apt/sources.list.d/debian.sources"
+
+        file_lines = host.file(source_file).content_string.split(os.linesep)
+        sources = [line for line in file_lines if line.startswith("deb")]
+
+        for source in sources:
+            source_elements = source.split(" ")
+            # The second element of a source entry can optionally be a
+            # set of options in the form "[options]".
+            repository_url = source_elements[
+                1 if not source_elements[1].startswith("[") else 2
+            ]
+            if host.system_info.distribution == "ubuntu" and (
+                "ubuntu.com" in repository_url or "canonical.com" in repository_url
+            ):
+                # We do not modify the standard repositories on Ubuntu systems
+                continue
+            assert repository_url.startswith("http://") is False
```

Comment:
Make sure you're okay with only asserting a failure when the repository URL starts with `http://`. If you only want to permit `https://`, you should assert that the repository URL starts with `https://`, which would exclude other protocols (like ftp).

Reply:
This role converts `http://` sources to `https://` so I believe this test makes sense. The only thing we want to guarantee is that there are no `http://` sources. If there were `ftp://` sources they should not be touched and would remain.

Reply:
The test is correct as written.  In the Ansible role we convert all repository URLs that start with `http://` to instead start with `https://` and that is what we are testing.